### PR TITLE
allow aria label updates on listview items

### LINKED ISSUE #206125: when navigating bw cells, indicate which cells are running in aria label
In the meeting with @jooyoungseo, Venkatesh suggested he'd like to know which cells are actively running when navigating with arrow keys.

Could we update the aria label to include "running" so that a user knows this?

### PR DESCRIPTION
**Pull request by amunger**
fix https://github.com/microsoft/vscode/issues/206125

This allows a list item update its aria label after being rendered by passing a callback to the accessibility provider.
@joaomoreno  - I'm very open to suggestions for how to improve the implementation here, but wanted to provide the concept of what I'm hoping to do.

Text that would be read by the screen reader: ("executing" is added when a cell is executing
image

### CODE REVIEW

**Review by rebornix — DISMISSED** [review]
Looks good to me in overall. I'll leave to @joaomoreno on if we want to go with event emitters (onDidChangeAria) or use update callbacks as used in this PR.

**Comment by joaomoreno** [comment]
I would definitely prefer if we were consistent with `onDidChangeActiveDescendant` and just have a matching event for `getAriaLabel(element: T)`.

```ts
interface IListAccessibilityProvider<T> {
  onDidChangeAriaLabel?: Event<T>;
}
```

https://github.com/microsoft/vscode/blob/5412743dcbeca03db7af33406bb541bb11d82cef/src/vs/base/browser/ui/list/listWidget.ts#L855

File: src/vs/base/browser/ui/list/listWidget.ts [hunk 1]
```diff
@@ -848,7 +851,8 @@ export interface IStyleController {
 }
 
 export interface IListAccessibilityProvider<T> extends IListViewAccessibilityProvider<T> {
-	getAriaLabel(element: T): string | null;
+	getAriaLabel(element: T, disposables?: DisposableStore): string | null;
```
Comment (joaomoreno) [inline src/vs/base/browser/ui/list/listWidget.ts:854]: Cool! No need to add this `disposables?` any more.

File: src/vs/base/browser/ui/list/listWidget.ts [hunk 2]
```diff
@@ -34,7 +34,10 @@ interface ITraitChangeEvent {
 	browserEvent?: UIEvent;
 }
 
-type ITraitTemplateData = HTMLElement;
+type ITraitTemplateData = {
+	container: HTMLElement;
+	disposables?: DisposableStore;
```
Comment (joaomoreno) [inline src/vs/base/browser/ui/list/listWidget.ts:39]: Should this just go into a separate `IAccessibilityTemplateData` type, and not be optional? We can then leave this type as is and have a much more concise change.

**Review by joaomoreno — CHANGES_REQUESTED** [review]

File: src/vs/base/browser/ui/list/listWidget.ts (hunk 2 above)
Comment (amunger) [reply src/vs/base/browser/ui/list/listWidget.ts:39]: sounds like that would help with the current typing errors

**Comment by amunger** [comment]
Still having trouble fixing these type issues with this new event on the Accessibility provider interface in the translation from Treeview options to List options. It's not clear if it's possible to get a TreeView item from a listView item.

**Comment by amunger** [comment]
https://github.com/microsoft/vscode/pull/207859/commits/5c6caf46b9a9339cb063a2a886160fa7b3f88a96 is the best I could come up with to satisfy the type system.

**Comment by joaomoreno** [comment]
Hmmm I see, unfortunate!

How about bringing in Observables here? https://github.com/microsoft/vscode/pull/208127

**Review by joaomoreno — APPROVED** [review]

**Pull request merged**